app/models/instance.rb

```diff
@@ -15,6 +15,9 @@ def find_tenant_by_host(host)
     end
   end
 
+  VALID_HOST_NAME_REGEX = /\A[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\z/i
+  validates :host, format: { with: VALID_HOST_NAME_REGEX }, length: { maximum: 255 }
```

Comment:
Dissued this with @fonglh 

It will be a good idea to think about whether using Regex or Gem would be better. In both cases, the readability of the code is still maintained so no issues here. 

@wangqiang1208 's proposed solution is rather straightforward and elegant, but I think as we catch some edge cases this regex might expand.

The value of implementing the gem here would be that i) there is a full set of tests available, which the gem is assumed to have passed (so you do not need to re-think about the test cases), ii) the validations provide error messages (see [here](https://github.com/KimNorgaard/validates_hostname/blob/master/lib/validates_hostname.rb))

Given the time pressure, may I suggest that we use this first. This will save @wangqiang1208 some time in designing the error messages. At some point, if need be, we will dig into the code to fix other issues.